[GTK][WPE][Skia] Refactor threaded CPU rendering code into its own class SkiaThreadedPaintingPool

nikolaszimmermann opened the pull request:
#### 6a5f94ce1b5592e092f9e9aa3070dd81c357e9a8
<pre>
[GTK][WPE][Skia] Refactor threaded CPU rendering code into its own class SkiaThreadedPaintingPool
<a href="https://bugs.webkit.org/show_bug.cgi?id=280589">https://bugs.webkit.org/show_bug.cgi?id=280589</a>

Reviewed by Carlos Garcia Campos.

This is a first step towards threaded GPU painting -- refactor the
threaded CPU rendering code (including display list recording tools)
into its own class: SkiaThreadedPaintingPool. Leave the GPU rendering
code as-is.

Covered by existing tests.

* Source/WebCore/platform/Skia.cmake:
* Source/WebCore/platform/SourcesSkia.txt:
* Source/WebCore/platform/graphics/GraphicsLayer.cpp:
(WebCore::GraphicsLayer::paintGraphicsLayerContents const):
(WebCore::GraphicsLayer::paintGraphicsLayerContents): Deleted.
* Source/WebCore/platform/graphics/GraphicsLayer.h:
(WebCore::GraphicsLayer::paintGraphicsLayerContents):
* Source/WebCore/platform/graphics/skia/SkiaThreadedPaintingPool.cpp: Added.
(WebCore::SkiaThreadedPaintingPool::SkiaThreadedPaintingPool):
(WebCore::SkiaThreadedPaintingPool::create):
(WebCore::SkiaThreadedPaintingPool::recordDisplayList const):
(WebCore::SkiaThreadedPaintingPool::postPaintingTask):
(WebCore::SkiaThreadedPaintingPool::numberOfPaintingThreads):
* Source/WebCore/platform/graphics/skia/SkiaThreadedPaintingPool.h: Added.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
(WebCore::CoordinatedGraphicsLayer::updateContentBuffers):
(WebCore::CoordinatedGraphicsLayer::paintIntoGraphicsContext const):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerCairo.cpp:
(WebCore::CoordinatedGraphicsLayer::paintTile):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp:
(WebCore::CoordinatedGraphicsLayer::paintTile):
* Source/WebCore/platform/graphics/texmap/coordinated/TiledBackingStore.h:
(WebCore::TiledBackingStore::contentsScale const):
(WebCore::TiledBackingStore::contentsScale): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.cpp:
(WebKit::CompositingCoordinator::CompositingCoordinator):
(WebKit::CompositingCoordinator::invalidate):
(WebKit::skiaNumberOfCpuPaintingThreads): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.h:

Canonical link: <a href="https://commits.webkit.org/284457@main">https://commits.webkit.org/284457@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e0812b0a931e49c23af9f04fe5f41ce5f9f090cb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69424 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48824 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22097 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73507 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20582 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/71541 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56625 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20433 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55220 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13680 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72490 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44541 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59939 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35698 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41208 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/17369 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18959 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63149 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17714 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75218 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13406 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16937 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62888 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13445 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/60022 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62794 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15412 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10809 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4423 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44628 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45702 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46897 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45443 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebCore/platform/graphics/skia/SkiaThreadedPaintingPool.cpp:
```diff
@@ -0,0 +1,130 @@
+
+SkiaThreadedPaintingPool::~SkiaThreadedPaintingPool()
+{
+}
```
```suggestion
SkiaThreadedPaintingPool::~SkiaThreadedPaintingPool() = default
```

nikolaszimmermann replied:
Fixed.

carlosgcampos reviewed (commented):
Why are we removing the non-threaded cpu rendering option? It was useful in the past to know if a particular issue was because of display lists or running on a different thread.

nikolaszimmermann committed:
[GTK][WPE][Skia] Refactor threaded CPU rendering code into its own class SkiaThreadedPaintingPool
https://bugs.webkit.org/show_bug.cgi?id=280589

Reviewed by Carlos Garcia Campos.

This is a first step towards threaded GPU painting -- refactor the
threaded CPU rendering code (including display list recording tools)
into its own class: SkiaThreadedPaintingPool. Leave the GPU rendering
code as-is.

Covered by existing tests.

* Source/WebCore/platform/Skia.cmake:
* Source/WebCore/platform/SourcesSkia.txt:
* Source/WebCore/platform/graphics/GraphicsLayer.cpp:
(WebCore::GraphicsLayer::paintGraphicsLayerContents const):
(WebCore::GraphicsLayer::paintGraphicsLayerContents): Deleted.
* Source/WebCore/platform/graphics/GraphicsLayer.h:
(WebCore::GraphicsLayer::paintGraphicsLayerContents):
* Source/WebCore/platform/graphics/skia/SkiaThreadedPaintingPool.cpp: Added.
(WebCore::SkiaThreadedPaintingPool::SkiaThreadedPaintingPool):
(WebCore::SkiaThreadedPaintingPool::create):
(WebCore::SkiaThreadedPaintingPool::recordDisplayList const):
(WebCore::SkiaThreadedPaintingPool::postPaintingTask):
(WebCore::SkiaThreadedPaintingPool::numberOfPaintingThreads):
* Source/WebCore/platform/graphics/skia/SkiaThreadedPaintingPool.h: Added.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
(WebCore::CoordinatedGraphicsLayer::updateContentBuffers):
(WebCore::CoordinatedGraphicsLayer::paintIntoGraphicsContext const):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerCairo.cpp:
(WebCore::CoordinatedGraphicsLayer::paintTile):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp:
(WebCore::CoordinatedGraphicsLayer::paintTile):
* Source/WebCore/platform/graphics/texmap/coordinated/TiledBackingStore.h:
(WebCore::TiledBackingStore::contentsScale const):
(WebCore::TiledBackingStore::contentsScale): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.cpp:
(WebKit::CompositingCoordinator::CompositingCoordinator):
(WebKit::CompositingCoordinator::invalidate):
(WebKit::skiaNumberOfCpuPaintingThreads): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.h:

Canonical link: https://commits.webkit.org/284457@main
